src/omap.rs

```diff
@@ -156,6 +188,122 @@ impl fmt::Debug for OMAPTable<'_> {
+}
+
+impl<'t> RangeIter<'t> {
+    /// Creates a `RangeIter` that does not yield any ranges.
+    pub fn empty() -> Self {
+        RangeIter {
+            records: [].iter(),
+            record: OMAPRecord::new(0, 0),
+            addr: 0,
+            end: 0,
+        }
+    }
+
+    /// Creates a `RangeIter` that only yields the specified range.
+    pub fn identity(start: u32, end: u32) -> Self {
+        // Declare the range `start..` as valid with an identity mapping. We cannot use `0..` here
+        // since the target must be a non-zero value to be recognized as valid mapping. Since there
+        // are no further records, a single subrange `start..end` will be considered.
+        RangeIter {
+            records: [].iter(),
+            record: OMAPRecord::new(start, start),
+            addr: start,
+            end,
+        }
+    }
+}
+
+impl Default for RangeIter<'_> {
+    fn default() -> Self {
+        Self::empty()
+    }
+}
+
+impl Iterator for RangeIter<'_> {
+    type Item = (u32, u32);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while self.addr < self.end {
+            // Pull the next record from the list. Since the current record is only valid up to the
+            // next one, this will determine the end of the current sub slice. If there are no more
+            // records, create an unmapped dummy record starting at the end of the source range.
+            let next_record = match self.records.next() {
+                Some(record) => *record,
+                None => OMAPRecord::new(self.end, 0),
+            };
+
+            // Calculate the bounds of the current subrange and write it back for the next
+            // iteration. Likewise, remember the next record as address translation base.
+            let subrange_end = cmp::min(next_record.source_address(), self.end);
+            let subrange_start = mem::replace(&mut self.addr, subrange_end);
+            let last_record = mem::replace(&mut self.record, next_record);
+
+            // Check for the validity of this sub-range or skip it silently:
+            //  2. The sub range covered by the last OMAP record might be empty. This can be an
+            //     artifact of a dummy record used when creating a new iterator.
+            //  3. A `target_address` of zero indicates an unmapped address range.
+            if subrange_start >= subrange_end || last_record.target_address() == 0 {
+                continue;
+            }
+
+            return Some((
+                last_record.translate(subrange_start),
+                last_record.translate(subrange_end),
+            ));
+        }
+
+        None
+    }
+}
+
+impl FusedIterator for RangeIter<'_> {}
+
+/// Iterator over [`Rva`] ranges returned by [`rva_ranges`].
+///
+/// [`Rva`]: struct.Rva.html
+/// [`rva_ranges`]: struct.AddressMap.html#method.rva_ranges
+pub struct RvaRangeIter<'t>(RangeIter<'t>);
+
+impl Iterator for RvaRangeIter<'_> {
+    type Item = (Rva, Rva);
```

Comment:
I'm _pretty sure_ this means `.0 <= addr < .1`, but I'm not completely sure. Please either document the semantics or swap out the tuple for e.g. `Range<Rva>` instead.

Reply:
Changed to `Range` now. The ranges methods on `AddressMap` could actually take `R: RangeBounds<_>`, however that makes the implementation of the iterator slighly more complicated. I didn't want to spend time on this for now - do you care to have this implemented?